src/services/category-service.ts

```diff
@@ -8,12 +8,22 @@ import {
   CategoriesParams,
   ItemsWithCount
 } from '~/types'
+import { baseService } from '~/services/base-service'
+import { getFullUrl } from '~/utils/helper-functions'
 
 export const categoryService = {
-  getCategories: (
-    params?: Partial<CategoriesParams>
-  ): Promise<AxiosResponse<ItemsWithCount<CategoryInterface>>> => {
-    return axiosClient.get(URLs.categories.get, { params })
+  getCategories: (params?: Partial<CategoriesParams>) => {
+    const { sort, ...rest } = params || {}
+
+    const resultUrl = getFullUrl({
+      pathname: URLs.categories.get,
+      searchParameters: { ...sort, ...rest }
+    })
```

Comment:
`getFullUrl` is compatible with nested objects like `sort`
If you will have a typescript error like `Index signature for type 'string' is missing in type`, just change the definition `CategoriesParams` from `interface` to a `type`.

```suggestion
  getCategories: (params?: Partial<CategoriesParams>) => {
    const resultUrl = getFullUrl({
      pathname: URLs.categories.get,
      searchParameters: params
    })
```

Reply:
That didn't really do the trick it still giving me the error